client_side/.infrastructure/infrastructure.sh

```diff
@@ -0,0 +1,351 @@
+      tellina_training
+    fi
+  fi
+}
+
+# Trains the user on how the infrastructure itself works. This includes:
+# - User meta-commands.
+# - Tasks and diff printing.
+# - The directory that they should be performing tasks on.
+infra_training() {
+  echo "--------------------------------------------------------------------------------"
+  echo "Please follow the instructions at"
+  echo "https://homes.cs.washington.edu/~atran35/research/bash_user_experiment/server_side/training/infrastructure_training.html"
+  echo "to learn how to do the experiment."
+}
+
+# Introduces the user to Tellina and suggests a couple of known query-command
+# pairs.
+tellina_training() {
+  echo "--------------------------------------------------------------------------------"
+  echo "This half of the experiment permits you to use the Tellina tool."
+  echo "Follow the instructions at"
+  echo "<tellina_training_url>"
+  echo "to acquaint yourself with Tellina:"
+}
+
+# Prints the list of resources that the user is allowed to use based on the
+# current treatment.
+print_treatment() {
+  if [[ "$treatment" == "T" ]]; then
+    echo "For this half of the experiment you can use any online resources and man pages,"
+    echo "and Tellina (https://tellina.rocks)."
+  else
+    if (( task_num >= TASKS_SIZE / 2 + 1 )); then
+      echo "For this half of the experiment you can use any online resources and man pages,"
+      echo "but you should not use Tellina."
+    else
+      echo "For this half of the experiment you can use any online resources and man pages."
+    fi
+  fi
+}
+
+# Prints the current task number and its description.
+print_task() {
+  echo "-----------------------------------------------------------------------"
+  echo "Task: ${task_num}/${TASKS_SIZE}"
+
+  ${INFRA_DIR}/jq-linux64 -r '.description' \
+    "${TASKS_DIR}/task_${task_code}/task_${task_code}.json"
+}
+
+# See documentation for ./verify_task.py for more details on what it does
+#
+# Parameters:
+# - $1: the directory that the user command was run in.
+#
+# Runs ./verify_task.py with the global task_code and the command in .command
+# and captures its exit code.
+#
+# If ./verify_task.py returns exit code:
+# - 0: sets status to success.
+# - 2: prints a prompt warning that the user has changed the file system and
+#      resets the file system.
+# - Otherwise: prints a prompt that the actual output does not match.
+#
+# Returns the exit code of ./verify_task.py
+verify_task() {
+  # Verify the output of the previous command.
+  local exit_code
+  local user_command="$(cat "${INFRA_DIR}/.command")"
+
+  "${INFRA_DIR}"/verify_task.py ${task_code} "$1" ${user_command}
+  exit_code=$?
+
+  case $exit_code in
+    0)
+      status="success"
+      ;;
+    2)
+      echo "You have modified the file system. It will now be reset to its" \
+        "original state."
+      reset_fs
+      ;&
+    *)
+      echo "Actual output does not match expected. A diff has been shown."
```

Comment:
This might be a message about the file system state.  I feel it's important to have different messages for wrong file system state and wrong output.

Reply:
Ah I misunderstood what you meant.